Firebase/CoreDiagnostics/ProtoSupport/Protos/firebasecore.proto

```diff
@@ -0,0 +1,102 @@
+syntax = "proto2";
+
+package logs.proto.mobilesdk.ios;
+
+message ICoreConfiguration {
+  enum ConfigurationType {
+    UNKNOWN_CONFIGURATION_TYPE = 0;
+    CORE = 1;
+    SDK = 2;
+  }
+  enum BuildType {
+    UNKNOWN_BUILD_TYPE = 0;
+    INTERNAL = 1;
+    EAP = 2;
+    PROD = 3;
+  }
+  enum ServiceType {
+    UNKNOWN_SDK_SERVICE = 0;
+    ICORE = 1;
+    ADMOB = 2;
+    APP_INVITE = 3;
+    SIGN_IN = 5;
+    GCM = 6 [deprecated = true];
+    MAPS = 7;
+    SCION = 8 [deprecated = true];
+    ANALYTICS = 9;
+    APP_INDEXING = 10;
+    CONFIG = 11 [deprecated = true];
+    DURABLE_DEEP_LINKS = 12 [deprecated = true];
+    CRASH = 13;
+    AUTH = 14;
+    DATABASE = 15;
+    STORAGE = 16;
+    MESSAGING = 17;
+    MEASUREMENT = 18;
+    REMOTE_CONFIG = 19;
+    DYNAMIC_LINKS = 20;
+    INVITES = 21;
+    AUTH_UI = 22;
+    FIRESTORE = 23;
+    PERFORMANCE = 24;
+    ML_VISION_ON_DEVICE_FACE = 26;
+    ML_VISION_ON_DEVICE_BARCODE = 27;
+    ML_VISION_ON_DEVICE_TEXT = 28;
+    ML_VISION_ON_DEVICE_LABEL = 29;
+    ML_MODEL_INTERPRETER = 30;
+    IN_APP_MESSAGING = 31;
+    FUNCTIONS = 32;
+    ML_NATURAL_LANGUAGE = 33;
+    ML_VISION_ON_DEVICE_AUTOML = 34;
+    ML_VISION_ON_DEVICE_OBJECT_DETECTION = 35;
+  }
+  enum PodName {
+    UNKNOWN_POD_NAME = 0;
+    GOOGLE = 1;
+    FIREBASE = 2;
+  }
+  enum DeploymentType {
+    UNKNOWN = 0;
```

Comment:
Is this an issue if the generated field name is `logs_proto_mobilesdk_ios_ICoreConfiguration_DeploymentType_UNKNOWN`?

Reply:
Oh, right. Nanopb just concatenates namespace pieces together since there's no namespacing in C. (Incidentally, in firestore, our proto generators *do* add namespaces to our nanopb files, mostly to cut down on the length and redundancy of these symbols.)

It's still an issue if you want to generate with libprotobuf for any other language, but otherwise it seems less important. (You *might* want to do this anyways, in case future you decides to use this proto for other languages.)

But if this is another team's proto, I'd personally try to make as few changes as possible (preferably none) to keep the files in sync. (Though you could try to upstream changes that you felt were especially important...)

Reply:
I'll just leave as is and let @davidair and @ryanwilson change this in the future if they so choose.